src/main/java/mate/jdbc/dao/impl/CarDaoImpl.java

```diff
@@ -0,0 +1,219 @@
+                + "WHERE c.id = ? AND c.is_deleted = FALSE;";
+
+        Optional<Car> car = Optional.empty();
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement
+                        = connection.prepareStatement(query, ResultSet.TYPE_SCROLL_INSENSITIVE,
+                        ResultSet.CONCUR_READ_ONLY)) {
+            statement.setLong(1, id);
+            ResultSet resultSet = statement.executeQuery();
+            if (resultSet.next()) {
+                car = Optional.of(getCar(resultSet));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get car by id: " + id, e);
+        }
+        return car;
+    }
+
+    private Car getCar(ResultSet resultSet) throws SQLException {
+        Manufacturer manufacturer = new Manufacturer();
+        manufacturer.setId(resultSet.getObject("manufacturer_id", Long.class));
+        manufacturer.setName(resultSet.getString("m.name"));
+        manufacturer.setCountry(resultSet.getString("m.country"));
+
+        Long carId = resultSet.getObject("id", Long.class);
+        String model = resultSet.getString("model");
+        List<Driver> drivers = new ArrayList<>();
+        int currentRow = resultSet.getRow();
+        do {
+            Long driverID = resultSet.getObject("d.id", Long.class);
+            String driverName = resultSet.getString("d.name");
+            String driverLicenseNumber = resultSet.getString("d.license_number");
+            drivers.add(new Driver(driverID, driverName, driverLicenseNumber));
+        } while (resultSet.next()
+                && carId.equals(resultSet.getObject("c.id", Long.class)));
+
+        resultSet.absolute(currentRow);
+        return new Car(carId, model, manufacturer, drivers);
+    }
+
+    @Override
+    public List<Car> getAll() {
+        String query = "SELECT c.*, m.id, m.name, m.country, d.id, d.name, d.license_number "
+                + "FROM " + CARS_TABLE + " c "
+                + "LEFT JOIN " + MANUFACTURERS_TABLE + " m ON c.manufacturer_id = m.id "
+                + "LEFT JOIN " + CARS_DRIVERS_TABLE + " cd ON c.id = cd.car_id "
+                + "LEFT JOIN " + DRIVERS_TABLE + " d ON cd.driver_id = d.id "
+                + "WHERE c.is_deleted = FALSE;";
+
+        List<Car> cars = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement
+                        = connection.prepareStatement(query, ResultSet.TYPE_SCROLL_INSENSITIVE,
+                        ResultSet.CONCUR_READ_ONLY)) {
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                if (!cars.isEmpty() && cars.get(cars.size() - 1).getId()
+                        .equals(resultSet.getObject("c.id", Long.class))) {
+                    continue;
+                }
+                cars.add(getCar(resultSet));
```

Comment:
Try this
```suggestion
                cars.add(getCar(resultSet));
                resultSet.previous();
```

Reply:
in `if()` condition I check if the last record in the `cars list` has the same `id` as the next row in `resultSet` - then go to next iteration (without calling the `getCar()` ), because this `car` had been allready recorded to the `list`, because the `getCar()` method takes the `car` and add all of the `drivers list` to the `car` at one(first) call.
So I don't see how  `resultSet.previous();` can help here.